### PR TITLE
Idaho 2023 tax rate

### LINKED ISSUE #3294: Idaho 2023 flat tax rate
in 2021 and 2022, Idaho has applied a progressive tax rate while in 2023 it switched to a [flat tax rate ](https://tax.idaho.gov/taxes/tax-pros/tax-update/tax-update-summer-2023/#:~:text=Idaho's%20tax%20rate%20is%20now,if%20you're%20filing%20jointly)of 5.8%

[House Bill](https://legislature.idaho.gov/sessioninfo/2023/legislation/H0172/)

### PR DESCRIPTION
**Pull request by jiajiagong0123**
Fixes #3294

### Summary
📝📊🎁

This pull request updates the income tax parameters for different filing statuses in Idaho, based on the latest data and legislation. It adds sources and documentation for the parameters, and implements the 2023 tax reform that simplifies and reduces the income tax rates. It affects the files `head_of_household.yaml`, `joint.yaml`, `separate.yaml`, `single.yaml`, and `widow.yaml` in the `policyengine_us/parameters/gov/states/id/tax/income/main` directory.

> _`income` tax changes_
> _Idaho updates brackets_
> _falling leaves of code_

### Walkthrough
*  Add sources of information for the income tax brackets for different filer types in Idaho ([link](https://github.com/PolicyEngine/policyengine-us/pull/3326/files?diff=unified&w=0#diff-90c13e7f0d9f4a7a5c966a01f164b232ce420aa2c4420cb088e0d25eedcb3734R19-R22), [link](https://github.com/PolicyEngine/policyengine-us/pull/3326/files?diff=unified&w=0#diff-55eedf5ab87c7c12dfb3548a65afc5fbd27b4c52735fdb22f864ccf3240ca7a2R14-R17), [link](https://github.com/PolicyEngine/policyengine-us/pull/3326/files?diff=unified&w=0#diff-b10e4d31fdd2197646f7f82956092e72f0c7db73ca3f8d61ee7d4deda6f44f8cR14-R17), [link](https://github.com/PolicyEngine/policyengine-us/pull/3326/files?diff=unified&w=0#diff-1f7afcd7c6d30092a0dba9ab478a7872dc3c9c33919bd219ee3e2c70d5c6a342R14-R17), [link](https://github.com/PolicyEngine/policyengine-us/pull/3326/files?diff=unified&w=0#diff-10a742b8b1887809cc0436ba04cf02bc29e7c07ba1cf5f8a7dc09961ef540a05R14-R17))
*  Update the income tax brackets for different filer types in Idaho according to the 2023 legislation, which lowers the first bracket rate to zero, increases the second bracket rate and threshold, and eliminates the third bracket ([link](https://github.com/PolicyEngine/policyengine-us/pull/3326/files?diff=unified&w=0#diff-90c13e7f0d9f4a7a5c966a01f164b232ce420aa2c4420cb088e0d25eedcb3734L25-R41), [link](https://github.com/PolicyEngine/policyengine-us/pull/3326/files?diff=unified&w=0#diff-55eedf5ab87c7c12dfb3548a65afc5fbd27b4c52735fdb22f864ccf3240ca7a2L19-R35), [link](https://github.com/PolicyEngine/policyengine-us/pull/3326/files?diff=unified&w=0#diff-b10e4d31fdd2197646f7f82956092e72f0c7db73ca3f8d61ee7d4deda6f44f8cL19-R35), [link](https://github.com/PolicyEngine/policyengine-us/pull/3326/files?diff=unified&w=0#diff-1f7afcd7c6d30092a0dba9ab478a7872dc3c9c33919bd219ee3e2c70d5c6a342L19-R35), [link](https://github.com/PolicyEngine/policyengine-us/pull/3326/files?diff=unified&w=0#diff-10a742b8b1887809cc0436ba04cf02bc29e7c07ba1cf5f8a7dc09961ef540a05L19-R35))

### CODE REVIEW

File: policyengine_us/parameters/gov/states/id/tax/income/main/head_of_household.yaml [hunk 1]
```diff
@@ -12,25 +12,31 @@ metadata:
       href: https://tax.idaho.gov/taxes/income-tax/individual-income/individual-income-tax-rate-schedule/
     - title: IDAHO STATUTES || Section 35.01.01.075 - TAX ON INDIVIDUALS, ESTATES, AND TRUSTS   
       href: https://casetext.com/regulation/idaho-administrative-code/title-idapa-35-tax-commission-state/rule-350101-income-tax-administrative-rules/section-350101075-tax-on-individuals-estates-and-trusts
-    - title: IDAHO STATUTES, TITLE 63, CHAPTER 30, SECTION 63-3024
-      href: https://legislature.idaho.gov/statutesrules/idstat/title63/t63ch30/sect63-3024/ 
     - title: LEGISLATIVE SESSIONS || 2022 EXTRAORDINARY LEGISLATIVE SESSION || 2022 EXTRAORDINARY SESSION LEGISLATIONHOUSE BILL 1
       href: https://legislature.idaho.gov/sessioninfo/2022spcl/legislation/H0001/
+    - title: IDAHO Tax Update for Summer 2023
+      href: https://tax.idaho.gov/taxes/tax-pros/tax-update/tax-update-summer-2023/#:~:text=Idaho's%20tax%20rate%20is%20now,if%20you're%20filing%20jointly
+    - title: IDAHO STATUTES, TITLE 63, CHAPTER 30, SECTION 63-3024
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/id/tax/income/main/head_of_household.yaml:19]: subparagraph

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/id/tax/income/main/head_of_household.yaml:19]: all yaml

File: policyengine_us/parameters/gov/states/id/tax/income/main/joint.yaml [hunk 2]
```diff
@@ -8,23 +8,29 @@ metadata:
     - title: Idaho State Tax Comission - Individual Income Tax Rate Schedule
       href: https://tax.idaho.gov/taxes/income-tax/individual-income/individual-income-tax-rate-schedule/
     - title: IDAHO STATUTES || Section 35.01.01.075 - TAX ON INDIVIDUALS, ESTATES, AND TRUSTS   
-      href: https://casetext.com/regulation/idaho-administrative-code/title-idapa-35-tax-commission-state/rule-350101-income-tax-administrative-rules/section-350101075-tax-on-individuals-estates-and-trusts
+      href: https://casetext.com/regulation/idaho-administrative-code/title-idapa-35-tax-commission-state/rule-350101-income-tax-administrative-rules/section-350101075-tax-on-individuals-estates-and-trusts 
+    - title: IDAHO Tax Update for Summer 2023
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/id/tax/income/main/joint.yaml:12]:
```suggestion
    - title: Idaho state tax commission 2023 tax update 
```

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/id/tax/income/main/joint.yaml:12]: all yaml

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/states/id/tax/income/main/joint.yaml [hunk 3]
```diff
@@ -8,23 +8,29 @@ metadata:
...
       2022-01-01: 3_324
+      2023-01-01: 8_978
     rate:
       2021-01-01: 0.031
       2022-01-01: 0.03
+      2023-01-01: 0.058
   - threshold:
       2021-01-01: 9_526
       2022-01-01: 9_974
+      2023-01-01: .inf
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/id/tax/income/main/joint.yaml:33]: do we need to set the thresholds to .inf ; are there integration tests for those cases?

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/id/tax/income/main/joint.yaml:33]: all yaml files

File: policyengine_us/parameters/gov/states/id/tax/income/main/widow.yaml [hunk 4]
```diff
@@ -9,22 +9,28 @@ metadata:
...
       2022-01-01: 3_324
+      2023-01-01: 8_978
     rate:
       2021-01-01: 0.031
       2022-01-01: 0.03
+      2023-01-01: 0.058
   - threshold:
       2021-01-01: 9_526
       2022-01-01: 9_974
+      2023-01-01: .inf
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/id/tax/income/main/widow.yaml:33]: same question as above

File: policyengine_us/parameters/gov/states/id/tax/income/main/single.yaml [hunk 5]
```diff
@@ -9,31 +9,40 @@ metadata:
...
-      2022-01-01: 8_311    
+      2022-01-01: 8_311
+      2023-01-01: .inf
     rate:
       2021-01-01: 0.055
       2022-01-01: 0.06
   - threshold:
       2021-01-01: 7_939
+      2022-01-01: .inf
+      2023-01-01: .inf
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/id/tax/income/main/single.yaml:46]:
```suggestion
```

File: policyengine_us/parameters/gov/states/id/tax/income/main/widow.yaml [hunk 6]
```diff
@@ -9,32 +9,40 @@ metadata:
...
       2022-01-01: 16_622
+      2023-01-01: .inf
     rate:
       2021-01-01: 0.055
       2022-01-01: 0.065
   - threshold:  
       2021-01-01: 15_878
-      2022-01-01: .inf  
+      2022-01-01: .inf
+      2023-01-01: .inf
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/id/tax/income/main/widow.yaml:46]:
```suggestion
```

File: policyengine_us/parameters/gov/states/id/tax/income/main/separate.yaml [hunk 7]
```diff
@@ -9,31 +9,40 @@ metadata:
...
       2021-01-01: 6_351
       2022-01-01: 8_311
+      2023-01-01: .inf
     rate:
       2021-01-01: 0.055
       2022-01-01: 0.06
   - threshold:
       2021-01-01: 7_939
+      2022-01-01: .inf
+      2023-01-01: .inf
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/id/tax/income/main/separate.yaml:46]:
```suggestion
```

File: policyengine_us/tests/policy/baseline/gov/states/id/tax/income/id_income_tax_before_non_refundable_credits.yaml [hunk 8]
```diff
@@ -70,3 +70,138 @@
...
+    id_income_tax_before_non_refundable_credits: 0
+
+- name: Separate filer with high income in 2023 # (100,000 - 4,489) * 0.058
+  period: 2023
+  input:
+    state_code: ID
+    id_taxable_income: 100_000
+    filing_status: SEPARATE
+  output:
+    id_income_tax_before_non_refundable_credits: 5539.638
```
Comment (PavelMakarchuk) [inline policyengine_us/tests/policy/baseline/gov/states/id/tax/income/id_income_tax_before_non_refundable_credits.yaml:207]:
```suggestion
    id_income_tax_before_non_refundable_credits: 5_539.638
```

File: policyengine_us/parameters/gov/states/id/tax/income/main/head_of_household.yaml [hunk 9]
```diff
@@ -12,30 +12,37 @@ metadata:
...
     - title: IDAHO STATUTES || Section 35.01.01.075 - TAX ON INDIVIDUALS, ESTATES, AND TRUSTS   
       href: https://casetext.com/regulation/idaho-administrative-code/title-idapa-35-tax-commission-state/rule-350101-income-tax-administrative-rules/section-350101075-tax-on-individuals-estates-and-trusts
-    - title: IDAHO STATUTES, TITLE 63, CHAPTER 30, SECTION 63-3024
-      href: https://legislature.idaho.gov/statutesrules/idstat/title63/t63ch30/sect63-3024/ 
     - title: LEGISLATIVE SESSIONS || 2022 EXTRAORDINARY LEGISLATIVE SESSION || 2022 EXTRAORDINARY SESSION LEGISLATIONHOUSE BILL 1
       href: https://legislature.idaho.gov/sessioninfo/2022spcl/legislation/H0001/
+    - title: Idaho state tax commission 2023 tax update
+      href: https://tax.idaho.gov/taxes/tax-pros/tax-update/tax-update-summer-2023/#:~:text=Idaho's%20tax%20rate%20is%20now,if%20you're%20filing%20jointly
+    - title: IDAHO STATUTES, TITLE 63, CHAPTER 30, SECTION 63-3024 (2)(a)
+      href: https://legislature.idaho.gov/statutesrules/idstat/title63/t63ch30/sect63-3024/
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/id/tax/income/main/head_of_household.yaml:20]: the law does not seem to align with our source, it states 5.8% over 2,500 or 5,000 

Which one is correct?

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/id/tax/income/main/head_of_household.yaml:20]: I think it should be 2,500 and 5,000 but please double check, that is what the law states

**Review by PavelMakarchuk — APPROVED** [review]

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**